src/vos/vos_internal.h

```diff
@@ -970,12 +974,19 @@ struct vos_rsrvd_scm {
 	struct pobj_action	*rs_actv;
 };
 
+D_CASSERT(offsetof(struct vos_rsrvd_scm, rs_actv_cnt) ==
+	  offsetof(struct dtx_rsrvd_scm, drs_cnt));
+D_CASSERT(offsetof(struct vos_rsrvd_scm, rs_actv_at) ==
+	  offsetof(struct dtx_rsrvd_scm, drs_at));
+D_CASSERT(offsetof(struct vos_rsrvd_scm, rs_actv) ==
+	  offsetof(struct dtx_rsrvd_scm, drs_ptr));
+D_CASSERT(sizeof(struct vos_rsrvd_scm) == sizeof(struct dtx_rsrvd_scm));
```

Comment:
It's better to use vos_rsrvd_scm only, and don't expose it, just use opaque pointer or array buffer if you need it in dtx_handle.

Reply:
'vos_rsrvd_scm' is vos internal data structure, I cannot expose it. But we need some way to attach it to the DTX handle, that is why I define another data structure in vos_types.h. If we use use an opaque pointer in DTX handle, then there is no efficient way to check its buffer size unless we record its size explicitly.